Add a CustomExecution for `swift` tasks

award999 opened the pull request:
To give us more control of the swift process and allow us to consume the output from swift, introduce our own CustomExecution. This will set us up to address #750, #653 and #694.

The CustomExecution will load the asar bundled node-pty module that ships with vscode. This way we don't need to build for every platform and arch

award999 commented:
I have some follow up commits to put up to make work on Windows and iron out some small stuff, but I'd appreciate any early feedback 😃

plemarquand commented on src/tasks/SwiftPseudoterminal.ts:
```diff
@@ -0,0 +1,91 @@
+    handleInput(data: string): void {
+        const buf: Buffer = Buffer.from(data);
+        // Kill on ctrl+c
+        if (buf.length === 1 && buf[0] === 3) {
```
Question, Is this called by vscode when you ctrl+c in the built in terminal? Is that the only way to receive this input sequence?

award999 replied:
Yep. VSCode calls handleInput, providing the buffer as a VT sequence. Can see https://vt100.net/docs/vt100-ug/chapter3.html. Look at Table 3-5  to see ctrl+c is 3

plemarquand reviewed (commented):
Looks good to me, just one question

award999 commented on src/tasks/SwiftExecution.ts:
```diff
@@ -0,0 +1,46 @@
+        public readonly command: string,
+        public readonly args: string[],
+        public readonly options: SwiftExecutionOptions
+    ) {
```
Are you thinking to make testing easier, i.e. can pass a stub?

adam-fowler replied:
Just for ease of use. As soon as this is available I have a use for it. Currently `swift package unedited` is run using execFile because I need to parse the output if it fails. I'd prefer to push this to a task, like all the other swift operations.

award999 replied:
So you would be able to pass in your own event listener to listen for output. I'm not sure we want to let overwriting the event itself. It may be more clear in the subsequent patches for diagnostics but this would be how you could accomplish:
```
const execution = task.execution;
if (!(execution && execution instanceof SwiftExecution)) {
      return;
}
const swiftExecution = execution as SwiftExecution;
swiftExecution.onDidWrite(data => {
     // do something with data
});

award999 commented on src/tasks/SwiftExecution.ts:
```diff
@@ -0,0 +1,46 @@
+
+    onDidClose: vscode.Event<number | void> = this.getSwiftProcess().onDidClose;
+
+    private getSwiftProcess(): SwiftProcess {
```
ACK

award999 commented on src/tasks/SwiftProcess.ts:
```diff
@@ -0,0 +1,94 @@
+                    ? false
+                    : undefined;
+            this.spawnedProcess = spawn(this.command, this.args, {
+                cwd: this.options.cwd,
```
ACK

award999 commented:
> I've added a couple of comments, plus can we have some function comment headers describing what everything is doing.

Added some more docstring comments

adam-fowler reviewed (approved):
I think we're good.
